### PR TITLE
update ipld spec with new link fmt

### PR DESCRIPTION
**Pull request by jbenet**
This changes the spec to update the link fmt. Please review it carefully and suggest changes. This was a great simplification. thanks everyone.

See:
- https://github.com/ipfs/specs/issues/97
- https://github.com/ipfs/js-ipld/pull/17

### CODE REVIEW

**Comment by dignifiedquire** [comment]
There is still a paragraph referencing `@link`:

> As a consequence of using the @link path component to denote cross-object traversals, this becomes a reserved path component and makes it impossible to access arbitrary @link keys that are not otherwise merkle-links. Escaping can be used to render access to those keys possible if so desired.

**Comment by dignifiedquire** [comment]
The section on `Escaping Algorithm` still refers to `@link`

File: ipld/README.md [hunk 1]
```diff
@@ -23,11 +25,45 @@ A _merkle-link_ is a link between two objects which is content-addressed with th
...
+```js
+{
+  "foo": {
+    "bar": "/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k", // not a link
+    "baz": {"/": "/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k"} // link
+  }
+}
 
-If an application wants to use the `@link` key for other purposes, the application itself is responsible to escape the keys in the IPLD object so that the application keys do not conflict with the `@link` key. When discussing application specific paths, it may be worth escaping all keys starting with `@` in case future versions of IPLD make use of other keys.
+**Object with pseudo "link object" at `files/cat.jpg` and actual link at `files/cat.jpg/link`**
```
Comment (dignifiedquire) [inline ipld/README.md]: This seems to be new to me and problematic when building it into the resolver. I was under the impression key like `"link"` would not have a special meaning, but with this you are effectively giving them one. 
Also this seems ambiguous to me, consider this object:

```js
{
  "files: {
    "cat.jpg": {
      "coolLink": {"/": "/ipfs/Qm..AAA"},
      "link": {"/": "/ipfs/QmBBB"},
      "mode": 0755
    }
  }
}
```

then how should `files/cat.jpg` be resolved? Objects are not inherently ordered in JavaScript so we can not guarantee the same resolution if we say first come, first serve.

Comment (jbenet) [inline ipld/README.md]: @dignifiedquire no, there's nothing special going on. the `link` keyword here is not special at all, it just shows how someone might model a data structure that requires associating properties with a link. Ex:

```
> ipld get .../files/cat.jpg
 {
  "coolLink": {"/": "/ipfs/Qm..AAA"},
  "link": {"/": "/ipfs/QmBBB"},
  "mode": 0755
}

> ipld get .../files/cat.jpg/coolLink/
<resolves through coolLink>

> ipld get .../files/cat.jpg/link/
<resolves through link>
```

Comment (dignifiedquire) [inline ipld/README.md]: Okay great, then I just misunderstood

**Comment by jbenet** [comment]
@dignifiedquire thanks, removed those `@link` refs.

**Pull request merged**